$Id: 098_en_US.UTF-8_Compose_fix_Unicode_plane_1.diff 471 2005-08-02 03:47:27Z dnusinow $

Jan Wilhelm Stumpel reported:
> In /usr/X11R6/lib/X11/locale/en_US.UTF-8/Compose, the Unicode 'Plane 1'
> characters (above Uffff, line 5577-5598) are damaged. E.g. the last
> character in the list, which should be U1D1C0 (hex f0 9d 87 80), is in
> reality UD1C0 (hex ed 87 80). Has this file spent some of its life in 16-bit
> Unicode form?

(See Debian #267321.)

This patch by Jan Wilhelm Stumpel corrects this problem.

Not submitted to XFree86.

Update: pt_BR.UTF-8 part removed. We replace it by a simple #include with a
couple of additional entries later anyway. (Eugene Konev)

Index: libx11-1.0.0/nls/en_US.UTF-8/Compose.pre
===================================================================
--- libx11-1.0.0.orig/nls/en_US.UTF-8/Compose.pre	2006-02-27 12:58:04.000000000 +0700
+++ libx11-1.0.0/nls/en_US.UTF-8/Compose.pre	2006-02-27 12:58:44.000000000 +0700
@@ -5576,25 +5576,25 @@
 <Multi_key> <U05bf> <hebrew_beth>	: "Ô≠å" UFB4C # HEBREW LETTER BET WITH RAFE
 <Multi_key> <U05bf> <hebrew_kaph>	: "Ô≠ç" UFB4D # HEBREW LETTER KAF WITH RAFE
 <Multi_key> <U05bf> <hebrew_pe>	: "Ô≠é" UFB4E # HEBREW LETTER PE WITH RAFE
-<Multi_key> <U1d157> <U1d165>	: "ÌÖû" U1D15E # MUSICAL SYMBOL HALF NOTE
-<Multi_key> <U1d158> <U1d165>	: "ÌÖü" U1D15F # MUSICAL SYMBOL QUARTER NOTE
-<Multi_key> <U1d15f> <U1d16e>	: "ÌÖ†" U1D160 # MUSICAL SYMBOL EIGHTH NOTE
-<Multi_key> <U1d158> <U1d165> <U1d16e>	: "ÌÖ†" U1D160 # MUSICAL SYMBOL EIGHTH NOTE
-<Multi_key> <U1d15f> <U1d16f>	: "ÌÖ°" U1D161 # MUSICAL SYMBOL SIXTEENTH NOTE
-<Multi_key> <U1d158> <U1d165> <U1d16f>	: "ÌÖ°" U1D161 # MUSICAL SYMBOL SIXTEENTH NOTE
-<Multi_key> <U1d15f> <U1d170>	: "ÌÖ¢" U1D162 # MUSICAL SYMBOL THIRTY-SECOND NOTE
-<Multi_key> <U1d158> <U1d165> <U1d170>	: "ÌÖ¢" U1D162 # MUSICAL SYMBOL THIRTY-SECOND NOTE
-<Multi_key> <U1d15f> <U1d171>	: "ÌÖ£" U1D163 # MUSICAL SYMBOL SIXTY-FOURTH NOTE
-<Multi_key> <U1d158> <U1d165> <U1d171>	: "ÌÖ£" U1D163 # MUSICAL SYMBOL SIXTY-FOURTH NOTE
-<Multi_key> <U1d15f> <U1d172>	: "ÌÖ§" U1D164 # MUSICAL SYMBOL ONE HUNDRED TWENTY-EIGHTH NOTE
-<Multi_key> <U1d158> <U1d165> <U1d172>	: "ÌÖ§" U1D164 # MUSICAL SYMBOL ONE HUNDRED TWENTY-EIGHTH NOTE
-<Multi_key> <U1d1b9> <U1d165>	: "ÌÜª" U1D1BB # MUSICAL SYMBOL MINIMA
-<Multi_key> <U1d1ba> <U1d165>	: "ÌÜº" U1D1BC # MUSICAL SYMBOL MINIMA BLACK
-<Multi_key> <U1d1bb> <U1d16e>	: "ÌÜΩ" U1D1BD # MUSICAL SYMBOL SEMIMINIMA WHITE
-<Multi_key> <U1d1b9> <U1d165> <U1d16e>	: "ÌÜΩ" U1D1BD # MUSICAL SYMBOL SEMIMINIMA WHITE
-<Multi_key> <U1d1bc> <U1d16e>	: "ÌÜæ" U1D1BE # MUSICAL SYMBOL SEMIMINIMA BLACK
-<Multi_key> <U1d1ba> <U1d165> <U1d16e>	: "ÌÜæ" U1D1BE # MUSICAL SYMBOL SEMIMINIMA BLACK
-<Multi_key> <U1d1bb> <U1d16f>	: "ÌÜø" U1D1BF # MUSICAL SYMBOL FUSA WHITE
-<Multi_key> <U1d1b9> <U1d165> <U1d16f>	: "ÌÜø" U1D1BF # MUSICAL SYMBOL FUSA WHITE
-<Multi_key> <U1d1bc> <U1d16f>	: "ÌáÄ" U1D1C0 # MUSICAL SYMBOL FUSA BLACK
-<Multi_key> <U1d1ba> <U1d165> <U1d16f>	: "ÌáÄ" U1D1C0 # MUSICAL SYMBOL FUSA BLACK
+<Multi_key> <U1d157> <U1d165>	: "ùÖû" U1D15E # MUSICAL SYMBOL HALF NOTE
+<Multi_key> <U1d158> <U1d165>	: "ùÖü" U1D15F # MUSICAL SYMBOL QUARTER NOTE
+<Multi_key> <U1d15f> <U1d16e>	: "ùÖ†" U1D160 # MUSICAL SYMBOL EIGHTH NOTE
+<Multi_key> <U1d158> <U1d165> <U1d16e>	: "ùÖ†" U1D160 # MUSICAL SYMBOL EIGHTH NOTE
+<Multi_key> <U1d15f> <U1d16f>	: "ùÖ°" U1D161 # MUSICAL SYMBOL SIXTEENTH NOTE
+<Multi_key> <U1d158> <U1d165> <U1d16f>	: "ùÖ°" U1D161 # MUSICAL SYMBOL SIXTEENTH NOTE
+<Multi_key> <U1d15f> <U1d170>	: "ùÖ¢" U1D162 # MUSICAL SYMBOL THIRTY-SECOND NOTE
+<Multi_key> <U1d158> <U1d165> <U1d170>	: "ùÖ¢" U1D162 # MUSICAL SYMBOL THIRTY-SECOND NOTE
+<Multi_key> <U1d15f> <U1d171>	: "ùÖ£" U1D163 # MUSICAL SYMBOL SIXTY-FOURTH NOTE
+<Multi_key> <U1d158> <U1d165> <U1d171>	: "ùÖ£" U1D163 # MUSICAL SYMBOL SIXTY-FOURTH NOTE
+<Multi_key> <U1d15f> <U1d172>	: "ùÖ§" U1D164 # MUSICAL SYMBOL ONE HUNDRED TWENTY-EIGHTH NOTE
+<Multi_key> <U1d158> <U1d165> <U1d172>	: "ùÖ§" U1D164 # MUSICAL SYMBOL ONE HUNDRED TWENTY-EIGHTH NOTE
+<Multi_key> <U1d1b9> <U1d165>	: "ùÜª" U1D1BB # MUSICAL SYMBOL MINIMA
+<Multi_key> <U1d1ba> <U1d165>	: "ùÜº" U1D1BC # MUSICAL SYMBOL MINIMA BLACK
+<Multi_key> <U1d1bb> <U1d16e>	: "ùÜΩ" U1D1BD # MUSICAL SYMBOL SEMIMINIMA WHITE
+<Multi_key> <U1d1b9> <U1d165> <U1d16e>	: "ùÜΩ" U1D1BD # MUSICAL SYMBOL SEMIMINIMA WHITE
+<Multi_key> <U1d1bc> <U1d16e>	: "ùÜæ" U1D1BE # MUSICAL SYMBOL SEMIMINIMA BLACK
+<Multi_key> <U1d1ba> <U1d165> <U1d16e>	: "ùÜæ" U1D1BE # MUSICAL SYMBOL SEMIMINIMA BLACK
+<Multi_key> <U1d1bb> <U1d16f>	: "ùÜø" U1D1BF # MUSICAL SYMBOL FUSA WHITE
+<Multi_key> <U1d1b9> <U1d165> <U1d16f>	: "ùÜø" U1D1BF # MUSICAL SYMBOL FUSA WHITE
+<Multi_key> <U1d1bc> <U1d16f>	: "ùáÄ" U1D1C0 # MUSICAL SYMBOL FUSA BLACK
+<Multi_key> <U1d1ba> <U1d165> <U1d16f>	: "ùáÄ" U1D1C0 # MUSICAL SYMBOL FUSA BLACK
